DSharpPlus/DiscordRestClient.cs

```diff
@@ -151,17 +151,16 @@ internal Task CreateGuildBanAsync(ulong guild_id, ulong user_id, int delete_mess
             if (delete_message_days < 0 || delete_message_days > 7)
                 throw new ArgumentException("Delete message days must be a number between 0-7", nameof(delete_message_days));
 
-            var pld = new RestGuildBanCreatePayload
-            {
-                DeleteMessageDays = delete_message_days,
-                Reason = reason
-            };
+            var urlparams = new Dictionary<string, string>();
+            urlparams["delete_message_days"] = delete_message_days.ToString();
```

Comment:
It's `delete-message-days`, dashes instead of underscores.

Also please use the initializer syntax where possible. It should look like this:

```cs
var urlparams = new Dictionary<string, string>()
{
    ["delete-message-days"] = delete_message_days.ToString()
};
```